bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/binding/firmware/Firmware.java

```diff
@@ -67,6 +68,7 @@
  * device is required afterwards) so that {@link FirmwareUpdateHandler}s can handle this information accordingly.
  *
  * @author Thomas Höfer - Initial contribution
+ * @author Dimitar Ivanov - FirmwareUID is replaced by ThingTypeUID and firmware version
```

Comment:
Please turn this class into an Interface and move the implementation into `org.eclipse.smarthome.core.thing.internal.firmware.FirmwareImpl`